From 2c342f38b1a7d8a5606b6e50d7ca1bc93e30d909 Mon Sep 17 00:00:00 2001
From: Liron Himi <lironh@marvell.com>
Date: Wed, 12 Dec 2018 13:39:01 +0200
Subject: [PATCH 8/8] crypto: inside-secure: add additional capabilities

Add the following capabilities:
	- EIP197_PROTOCOL_HASH_ENCRYPT
	- EIP197_PROTOCOL_DECRYPT_HASH
	- EIP197_ALG_AES_OFB
	- EIP197_ALG_AES_CFB
	- EIP197_ALG_DES_OFB
	- EIP197_ALG_DES_CFB
	- EIP197_ALG_3DES_OFB
	- EIP197_ALG_3DES_CFB
	- EIP197_ALG_AES_XCBC_MAC
	- EIP197_ALG_AES_CTR_ICM
	- EIP197_ALG_GCM_HASH

Signed-off-by: Liron Himi <lironh@marvell.com>
---
 drivers/crypto/inside-secure/safexcel.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/crypto/inside-secure/safexcel.c b/drivers/crypto/inside-secure/safexcel.c
index 5521886..6979797 100644
--- a/drivers/crypto/inside-secure/safexcel.c
+++ b/drivers/crypto/inside-secure/safexcel.c
@@ -460,12 +460,19 @@ static int safexcel_hw_init(struct safexcel_crypto_priv *priv)
 		val = EIP197_FUNCTION_RSVD;
 		val |= EIP197_PROTOCOL_ENCRYPT_ONLY | EIP197_PROTOCOL_HASH_ONLY;
 		val |= EIP197_PROTOCOL_ENCRYPT_HASH | EIP197_PROTOCOL_HASH_DECRYPT;
+		val |= EIP197_PROTOCOL_HASH_ENCRYPT |
+			EIP197_PROTOCOL_DECRYPT_HASH;
 		val |= EIP197_ALG_DES_ECB | EIP197_ALG_DES_CBC;
+		val |= EIP197_ALG_DES_OFB | EIP197_ALG_DES_CFB;
 		val |= EIP197_ALG_3DES_ECB | EIP197_ALG_3DES_CBC;
+		val |= EIP197_ALG_3DES_OFB | EIP197_ALG_3DES_CFB;
 		val |= EIP197_ALG_AES_ECB | EIP197_ALG_AES_CBC;
+		val |= EIP197_ALG_AES_OFB | EIP197_ALG_AES_CFB;
 		val |= EIP197_ALG_MD5 | EIP197_ALG_HMAC_MD5;
 		val |= EIP197_ALG_SHA1 | EIP197_ALG_HMAC_SHA1;
 		val |= EIP197_ALG_SHA2 | EIP197_ALG_HMAC_SHA2;
+		val |= EIP197_ALG_AES_CTR_ICM | EIP197_ALG_GCM_HASH;
+		val |= EIP197_ALG_AES_XCBC_MAC;
 		writel(val, EIP197_PE(priv) + EIP197_PE_EIP96_FUNCTION_EN(pe));
 	}
 
-- 
2.7.4

